File: leetcode/algorithms/binary-search/binary-search.go
func search(nums []int, target int) int {
    
    var low int = 0
    var high int = len(nums)-1
    
    var sol int = -1;
    for(high-low>1){
        var mid int = low + ((high - low) / 2)
        
        if(nums[mid]==target){
            sol = mid
            break
        } else if(nums[mid]>target){
            high = mid
        } else {
            low = mid
        }
    }
    
    if(nums[low]==target){
        sol = low
    } else if(nums[high]==target){
        sol = high
    }
    
    return sol
    
}

File: leetcode/algorithms/binary-search/binary-search.ts
function search(nums: number[], target: number): number {
    
    let targetIndex = -1, l=0,h=nums.length-1;
    
    while(h>=l){
        const m = l + ~~((h-l)/2);
        if(nums[m]===target){
            targetIndex = m;
            break;
        } else if(nums[m]>target){
            h=m-1;
        } else {
            l=m+1;
        }
    }
    
    return targetIndex;

};